password/impl/src/test/java/org/wildfly/security/password/impl/MaskedPasswordPicketBoxCompatibilityTest.java

```diff
@@ -0,0 +1,96 @@
+
+import java.security.Provider;
+import java.security.Security;
+import java.security.spec.KeySpec;
+import java.util.Arrays;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wildfly.security.WildFlyElytronProvider;
+import org.wildfly.security.password.PasswordFactory;
+import org.wildfly.security.password.interfaces.MaskedPassword;
+import org.wildfly.security.password.spec.ClearPasswordSpec;
+import org.wildfly.security.password.spec.EncryptablePasswordSpec;
+import org.wildfly.security.password.spec.IteratedSaltedPasswordAlgorithmSpec;
+import org.wildfly.security.password.spec.MaskedPasswordSpec;
+
+/*
+ * @author Jan Kalina <jkalina@redhat.com>
+ * @author Zoran Regvart <zregvart@redhat.com>
+ */
+public class MaskedPasswordPicketBoxCompatibilityTest {
+
+    private final char[] initialKey = "somearbitrarycrazystringthatdoesnotmatter".toCharArray();
+
+    private final String picketBoxAlgorithm = MaskedPassword.ALGORITHM_MASKED_MD5_DES;
+
+    private final int picketBoxIterationCount = 50;
+
+    private final byte[] picketBoxMaskedPasswordBytes = new byte[] {-23, -52, 73, 36, -27, -114, 50, -81};
+
+    private final byte[] picketBoxSalt = "Mxyzptlk".getBytes();
+
+    private final String secret = "vault13";
+
+    private static final Provider provider = new WildFlyElytronProvider();
+
+    @BeforeClass
+    public static void setup() {
+        Security.addProvider(provider);
+    }
+
+    @AfterClass
+    public static void removeProvider() {
+        Security.removeProvider(provider.getName());
+    }
+
+    @Test
+    public void shouldBeCompatibleWithPicketBoxMaskedPasswords() throws Exception {
+        PasswordFactory factory = PasswordFactory.getInstance(picketBoxAlgorithm);
+
+        KeySpec maskedSpec = new MaskedPasswordSpec(initialKey, picketBoxIterationCount, picketBoxSalt, picketBoxMaskedPasswordBytes);
+        MaskedPassword picketBoxMaskedPassword = (MaskedPassword) factory.generatePassword(maskedSpec);
+
+        ClearPasswordSpec clearPasswordSpec = factory.getKeySpec(picketBoxMaskedPassword, ClearPasswordSpec.class);
+
+        assertEquals("Masked password should unmask PicketBox masked password", secret,
+                new String(clearPasswordSpec.getEncodedPassword()));
+    }
+
+    @Test
+    public void shouldGenerateMaskedPasswordsAsInPicketbox() throws Exception {
+        PasswordFactory factory = PasswordFactory.getInstance(picketBoxAlgorithm);
+
+        KeySpec passwordSpec = new EncryptablePasswordSpec(secret.toCharArray(),
+                new IteratedSaltedPasswordAlgorithmSpec(picketBoxIterationCount, picketBoxSalt));
+        MaskedPassword maskedPassword = (MaskedPassword) factory.generatePassword(passwordSpec);
+
+        byte[] encoded = maskedPassword.getMaskedPasswordBytes();
+
+        assertTrue("Password masking should generate same values as PicketBox",
+                Arrays.equals(picketBoxMaskedPasswordBytes, encoded));
+    }
+}
```

Comment:
This is a non issue, but if you use IntelliJ IDEA, there is a setting to insert new lines at the end of files (Settings → Editor → General → Ensure line feed at file end on Save ).

Reply:
Thanks, I've changed the setting